src/app/shared/pagination/pagination.directive.ts

```diff
@@ -0,0 +1,130 @@
+import {
+    Directive,
+    inject,
+    Input,
+    OnChanges,
+    SimpleChanges,
+    TemplateRef,
+    ViewContainerRef,
+} from '@angular/core';
+import {BehaviorSubject, filter, map} from 'rxjs';
+
+interface PaginationDirectiveContext<T> {
+    $implicit: T[];
+    index: number;
+    pageIndexes: number[];
+    appPaginationOf: T[];
+    next: () => void;
+    back: () => void;
+    selectIndex: (index: number) => void;
+}
+
+@Directive({
+    selector: '[appPagination]',
+})
+export class PaginationDirective<T> implements OnChanges {
+    private readonly viewContainerRef = inject(ViewContainerRef);
+    private readonly templateRef = inject<TemplateRef<PaginationDirectiveContext<T>>>(TemplateRef);
+
+    private readonly currentIndex$ = new BehaviorSubject<number>(0);
+
+    @Input() appPaginationOf: T[] | null | undefined;
+    @Input() appPaginationChunkSize: number | undefined;
```

Comment:
Я бы рекомендовал тут задать дефолтное значение, чтобы не работать в дальнейшем с undefined
```suggestion
    @Input() appPaginationChunkSize = 1;
```